servant-auth-server/src/Servant/Auth/Server/Internal/ConfigTypes.hs

```diff
@@ -33,18 +35,19 @@ data JWTSettings = JWTSettings
   -- | Algorithm used to sign JWT.
   , jwtAlg          :: Maybe Jose.Alg
   -- | Keys used to validate JWT.
-  , validationKeys  :: Jose.JWKSet
+  , validationKeys  :: IORef Jose.JWKSet
   -- | An @aud@ predicate. The @aud@ is a string or URI that identifies the
   -- intended recipient of the JWT.
   , audienceMatches :: Jose.StringOrURI -> IsMatch
   } deriving (Generic)
 
 -- | A @JWTSettings@ where the audience always matches.
-defaultJWTSettings :: Jose.JWK -> JWTSettings
-defaultJWTSettings k = JWTSettings
+defaultJWTSettings :: Jose.JWK -> IO JWTSettings
```

Comment:
I was thinking we could even omit the IO via `unsafePerformIO` to avoid forcing existing users change their code. It should be relatively safe as the only thing we're doing is create a new `IORef`.

This will make the changes even less invasive (or in most cases invisible).

I guess the library maintainers should decide on that.